stages/org.osbuild.qemu

```diff
@@ -24,13 +24,19 @@ SCHEMA_2 = r"""
     "description": "Create a qcow2 image",
     "type": "object",
     "required": ["type"],
+    "not": {"required": ["disable-compression", "compat"]},
     "additionalProperties": false,
     "properties": {
       "type": {
         "description": "The type of the format, here 'qcow2'",
         "type": "string",
         "enum": ["qcow2"]
       },
+      "disable-compression": {
+        "description": "Disable compression of the qcow2 image",
+        "type": "boolean",
+        "enum": [true]
+      },
```

Comment:
The current schema permits all combinations of `disable-compression` and `compat` options, except when they are both specified (characteristic of a NAND gate). Preventing both options from being specified means we don't have to deal with the case where `disable-compression = false` and `compat` should technically still be able to be specified. 

To cover the case where `disable-compression = false` and `compat` is not specified, I've forced `disable-compression` to only accept `true` as a value. This just makes the schema cleaner. However it is possible to create a more comprehensive schema, as seen with the deployment_path() optimization PR: https://github.com/osbuild/osbuild/blob/ef7f48e0114a028fec1021315b01f7142c113e96/mounts/org.osbuild.ostree.deployment#L41-L63

@dustymabe curious to hear your thoughts.

Reply:
Why do we need to make the options exclusive? Why can't someone specify compat and compression options?

Reply:
Oh okay, my bad. I thought that `compat` defined the type/version of compression used, not that it was something completely different. I'll fix this up.